From: Lukas Märdian <lukas.maerdian@oss.qualcomm.com>
Date: Mon, 14 Sep 2026 16:00:00 +0200
Subject: Build the direct AGM API integration test

Build the direct AGM API test as part of the top-level Autotools project.

Forwarded: not-needed
---
 Makefile.am              | 4 ++--
 configure.ac             | 1 +
 service/test/Makefile.am | 11 ++++-------
 3 files changed, 7 insertions(+), 11 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 8e4a7e2..d3b8b29 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,5 +1,5 @@
 if AGM_NO_IPC
-SUBDIRS = snd_parser service plugins/tinyalsa plugins/tinyalsa/test plugins/alsalib
+SUBDIRS = snd_parser service service/test plugins/tinyalsa plugins/tinyalsa/test plugins/alsalib
 else
-SUBDIRS = snd_parser service ipc/DBus/agm_client ipc/DBus/agm_server plugins/tinyalsa plugins/tinyalsa/test plugins/alsalib
+SUBDIRS = snd_parser service service/test ipc/DBus/agm_client ipc/DBus/agm_server plugins/tinyalsa plugins/tinyalsa/test plugins/alsalib
 endif
diff --git a/configure.ac b/configure.ac
index 770fd5a..5cb1f9e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -86,6 +86,7 @@ AC_CONFIG_FILES(
 Makefile
 service/Makefile
 service/agm.pc
+service/test/Makefile
 ipc/DBus/agm_client/Makefile
 ipc/DBus/agm_client/agmclient.pc
 ipc/DBus/agm_server/Makefile
diff --git a/service/test/Makefile.am b/service/test/Makefile.am
index 5ef0fe2..15b9b6d 100644
--- a/service/test/Makefile.am
+++ b/service/test/Makefile.am
@@ -1,10 +1,5 @@
-AM_CPPFLAGS := -I $(PKG_CONFIG_SYSROOT_DIR)/usr/include/agm/ -I ${top_srcdir}/src
-bin_PROGRAMS :=  agm_ipc_test
-agm_ipc_test_SOURCES   = ${top_srcdir}/src/agm_test.c
-agm_ipc_test_CPPFLAGS := $(AM_CPPFLAGS)
-agm_ipc_test_LDADD    = -lagmclient
+AM_CPPFLAGS = -I$(top_srcdir)/service/inc/public
+bin_PROGRAMS = agmtest
+agmtest_SOURCES = $(top_srcdir)/service/test/src/agm_test.c
+agmtest_LDADD = $(top_builddir)/service/libagm.la
 
-bin_PROGRAMS +=  agmtest
-agmtest_SOURCES   = ${top_srcdir}/src/agm_test.c
-agmtest_CPPFLAGS := $(AM_CPPFLAGS)
-agmtest_LDADD    = -lagm
